app/web/features/search/FloatingSearchControls.tsx

```diff
@@ -0,0 +1,298 @@
+}
+
+const StyledControlsWrapper = styled("div")(({ theme }) => ({
+  display: "flex",
+  alignItems: "center",
+}));
+
+const StyledButtonsContainer = styled("div")(({ theme }) => ({
+  display: "flex",
+  position: "relative",
+  fontSize: " 14px",
+  alignItems: "center",
+  justifyContent: "space-between",
+  height: "auto",
+  backgroundColor: theme.palette.common.white,
+  borderRadius: "50px",
+  boxShadow: theme.shadows[4],
+  width: "100%",
+  padding: theme.spacing(0.5, 1),
+}));
+
+const StyledFlexRow = styled("div")({
+  display: "flex",
+  alignItems: "center",
+  justifyContent: "space-between",
+});
+
+const sharedInputStyles = () => ({
+  height: "40px",
+  minWidth: "250px",
+  maxWidth: "250px",
+  marginLeft: theme.spacing(1),
+
+  "& .MuiInputBase-root": {
+    height: "40px",
+    minWidth: "250px",
+    maxWidth: "250px",
+    padding: 0,
+  },
+
+  "& .MuiInputBase-input": {
+    height: "40px",
+    padding: 0,
+  },
+
+  "& .MuiOutlinedInput-notchedOutline": {
+    border: "none",
+  },
+});
+
+const StyledLocationAutocompleteOutlined = styled(LocationAutocompleteOutlined)(
+  sharedInputStyles,
+);
+
+const StyledTextField = styled(TextField)(({ theme }) => ({
+  ...sharedInputStyles(),
+
+  "& .MuiInputBase-input": {
+    padding: 0,
+    height: "40px",
+  },
+}));
+
+const StyledSelect = styled(Select)(({ theme }) => ({
+  backgroundColor: "white",
+  borderRadius: "100px",
+  border: "none",
+  padding: theme.spacing(1),
+  display: "flex",
+  alignItems: "center",
+  height: "40px", // Match height with LocationAutocompleteOutlined
+  width: theme.spacing(15),
+
+  "& .MuiSelect-select": {
+    padding: 0,
+  },
+
+  "& .MuiOutlinedInput-notchedOutline": {
+    border: "none",
+  },
+
+  "& .Mui-focused .MuiOutlinedInput-notchedOutline": {
+    border: "none",
+  },
+}));
+
+const StyledTuneIcon = styled(Tune, {
+  shouldForwardProp: (prop) => prop !== "hasActiveFilters",
+})<{ hasActiveFilters: boolean }>(({ theme, hasActiveFilters }) => ({
+  color: hasActiveFilters
+    ? theme.palette.primary.main
+    : theme.palette.grey[500],
+  fontSize: "38px",
+  cursor: "pointer",
+  height: "20px",
+  width: "20px",
+  padding: 0,
+}));
+
+const StyledClearIcon = styled(Clear)(({ theme }) => ({
+  color: theme.palette.grey[500],
+  fontSize: "30px",
+  paddingRight: theme.spacing(1),
+  height: "20px",
+  width: "20px",
+  padding: 0,
+
+  "&:hover": {
+    cursor: "pointer",
+    color: theme.palette.primary.dark,
+  },
+}));
+
+const FloatingSearchControls = ({
+  mapRef,
+  onOpenFilters,
+  onSetSearchType,
+  searchType,
+}: FloatingSearchNavigationProps) => {
+  const { t } = useTranslation([SEARCH]);
+  const isMobile = useMediaQuery(theme.breakpoints.down("md"));
+
+  const [keyword, setKeyword] = useState("");
+
+  const {
+    search: { query },
+    hasActiveFilters,
+  } = useMapSearchState();
+
+  const { setSearch, clearSearchFilters, clearSearchInputValue } =
+    useMapSearchActions();
```

Comment:
this hooks makes it look so better

Reply:
I think there's still room for improvement if I adjust the component tree structure but it definitely clears up a lot of clutter!